tests/test_data_model.py

```diff
@@ -71,13 +71,30 @@ def test_lstm_simple(self):
         # Create from model
         LSTMStateTransitionModel(m.model, output_keys = ['x'])
 
+        # Test pickling model m
+        import pickle
```

Comment:
Suggest to move to the top of the file.

Reply:
This could be a Windows issue, but I also know there are caveats with pickling keras models and I actually have never pickled a keras model - always used the built in functions. 
 
I get the following error for test_lstm_simple:

```
======================================================================
ERROR: test_lstm_simple (__main__.TestDataModel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<ipython-input-1-e2dee28b68ec>", line 79, in test_lstm_simple
    m2 = pickle.loads(pickled_m)
  File "C:\Users\darrahts\anaconda3\envs\tf2x\lib\site-packages\keras\saving\pickle_utils.py", line 48, in deserialize_model_from_bytecode
    model = save_module.load_model(temp_dir)
  File "C:\Users\darrahts\anaconda3\envs\tf2x\lib\site-packages\keras\utils\traceback_utils.py", line 67, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\darrahts\anaconda3\envs\tf2x\lib\site-packages\tensorflow\python\saved_model\load.py", line 978, in load_internal
    str(err) + "\n You may be trying to load on a different device "
FileNotFoundError: Unsuccessful TensorSliceReader constructor: Failed to find any matching files for ram://5766af8a-251b-4258-a347-f22418e2ada6/variables/variables
 You may be trying to load on a different device from the computational device. Consider setting the `experimental_io_device` option in `tf.saved_model.LoadOptions` to the io_device such as '/job:localhost'.

----------------------------------------------------------------------
Ran 4 tests in 39.067s

FAILED (errors=1)
```

references:

https://github.com/tensorflow/tensorflow/issues/34697
https://stackoverflow.com/questions/73294899/unable-to-load-tensorflow-model-with-pickle

Reply:
Very interested- it seems to work fine on Mac and Linux. 

Since pickling doesn't seem to work well with keras models, I added a warning which should now come up when anyone tries to pickle the LSTMStateTransitionModel